service/get.go

```diff
@@ -0,0 +1,77 @@
+package service
+
+import (
+	"io/ioutil"
+	"log"
+
+	"k8s.io/helm/pkg/getter"
+	"k8s.io/helm/pkg/helm/environment"
+	"k8s.io/helm/pkg/repo"
+)
+
+// GetServiceInterface defines a Get service
+type GetServiceInterface interface {
+	Get() error
+}
+
+// GetService structure definition
+type GetService struct {
+	config       repo.Entry
+	verbose      bool
+	ignoreErrors bool
+	logger       *log.Logger
+}
+
+// NewGetService return a new instace of GetService
+func NewGetService(config repo.Entry, verbose bool, ignoreErrors bool, logger *log.Logger) GetServiceInterface {
+	return &GetService{
+		config:       config,
+		verbose:      verbose,
+		ignoreErrors: ignoreErrors,
+		logger:       logger,
+	}
+}
+
+//Get methods downloads the index file and the Helm charts to the working directory.
+func (g *GetService) Get() error {
+	chartRepo, err := repo.NewChartRepository(&g.config, getter.All(environment.EnvSettings{}))
+	if err != nil {
+		return err
+	}
+
+	err = chartRepo.DownloadIndexFile(g.config.Name + "/downloaded-index.yaml")
+	if err != nil {
+		return err
+	}
+
+	err = chartRepo.Load()
+	if err != nil {
+		return err
+	}
+
+	charts := chartRepo.IndexFile.Entries
+	for n, c := range charts {
+		for _, cc := range c {
+			for _, u := range cc.URLs {
+				b, err := chartRepo.Client.Get(u)
+				if err != nil {
+					return err
+				}
+				err = writeFile(g.config.Name+"/"+n+"-"+cc.Version+".tgz", b.Bytes(), g.logger)
+				if err != nil {
+					return err
```

Comment:
I have to admit that I'm not fully aware of what cc.URLs is expected to contain, but isn't a bit harsh to return early on an error of a single error? I'm thinking, maybe other URLs do work, and we have to report only one download error (in which case we'd return an error anyways, but at least other charts would have been downloaded, which is specially useful for big repositories I guess).

Reply:
Thanks for the hint here will add that mechanism.